### PR TITLE
[5.1] Fix exists for SqlServer

### PR DESCRIPTION
**Pull request by schinken**
Fixed broken exists statement as discussed in #10373

### CODE REVIEW

File: src/Illuminate/Database/Query/Grammars/SqlServerGrammar.php [hunk 1]
```diff
@@ -211,9 +211,10 @@ public function compileTruncate(Builder $query)
      */
     public function compileExists(Builder $query)
     {
-        $select = $this->compileSelect($query);
+        $query->selectRaw('1 [exists]')
+              ->limit(1);
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Grammars/SqlServerGrammar.php]: Please keep this one line line

Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Grammars/SqlServerGrammar.php]: `$query->selectRaw('1 [exists]')->limit(1);`

**Comment by schinken** [comment]
@GrahamCampbell I fixed the code as you mentioned in your comment and updated by pull request

**Comment by yajra** [comment]
@schinken I just realized that columns should be reset when implementing the exists queries. See this commit that I've made for ref: https://github.com/yajra/laravel-oci8/commit/6452860e46514ca381cdb92cc72e755b403a0975

**Comment by schinken** [comment]
@yajra What's the reason for cloning the Builder object? I also thought about resetting the columns...

**Comment by yajra** [comment]
It's for duplicating the query object just to be sure that we will not be modifying the existing query when we reset the columns to empty array.

**Comment by schinken** [comment]
@yajra Check my updated commit

**Comment by yajra** [comment]
@schinken looks good for me. :+1:

**Comment by taylorotwell** [comment]
Can anyone that uses SQL Server confirms that this actually works?

**Comment by schinken** [comment]
I've tested the query on an SQL Server before making this commit:

```bash
christopher@cucumber:~$ sqlcmd -S tcp:sqlsrv -U "testuser" -Q "select top 1 1 [exists] from [users] WHERE id = 1"
exists
-----------
          1

(1 rows affected)
christopher@cucumber:~$ sqlcmd -S tcp:sqlsrv -U "testuser" -Q "select top 1 1 [exists] from [users] WHERE id = 32498273497"
exists
-----------

(0 rows affected)
```

**Pull request merged**